app/src/main/java/com/example/peterchu/watplanner/home/HomePresenter.java

```diff
@@ -68,4 +80,62 @@ public boolean onSubmitSearchQuery(String subject, String catalogNumber) {
         Course c = dataRepository.getCourse(subject, catalogNumber);
         return homeFragment.openDetailView(c == null ? null : c.getId());
     }
+
+    public void onExportCoursesToCalendar(List<CourseComponent> courseSchedule) {
+        for (CourseComponent courseComponent : courseSchedule) {
+            long calID = getDefaultCalendarID(this.homeFragment.getActivity());
+            Calendar beginTime = Calendar.getInstance();
+            beginTime.set(2016, 0, 19, 7, 30);
+            long startMillis = beginTime.getTimeInMillis();
+            Calendar endTime = Calendar.getInstance();
+            endTime.set(2016, 0, 19, 8, 30);
+            long endMillis = endTime.getTimeInMillis();
+
+            ContentResolver cr = this.homeFragment.getActivity().getContentResolver();
+            ContentValues values = new ContentValues();
+            values.put(CalendarContract.Events.DTSTART, startMillis);
+            values.put(CalendarContract.Events.DTEND, endMillis);
+            values.put(CalendarContract.Events.TITLE, String.format("%s %d",
+                    courseComponent.getSubject(), courseComponent.getCatalogNumber()));
+            values.put(CalendarContract.Events.DESCRIPTION, courseComponent.getTitle());
+            values.put(CalendarContract.Events.CALENDAR_ID, calID);
+            values.put(CalendarContract.Events.EVENT_TIMEZONE, "America/Los_Angeles");
+
+            boolean canWriteCalendar = ContextCompat.checkSelfPermission(
+                    this.homeFragment.getActivity(),
+                    Manifest.permission.WRITE_CALENDAR
+            ) == PackageManager.PERMISSION_GRANTED;
+
+            if (!canWriteCalendar) {
```

Comment:
I believe it's when a user hasn't given the app permission previously. It's been a thing since Android 6.0, where permissions are checked and granted at run-time